staff/Leticia/js/arrays.js

```diff
@@ -0,0 +1,816 @@
+
+//     Input: numbers= [10,20,10,40,50,60,70], target=50
+//     Output: 3, 4
+
+
+
+// 9 ### validValues
+// Write a JavaScript function to retrieve the value of a given property from all elements in an array.
+
+//     Sample array : [NaN, 0, 15, false, -22, '',undefined, 47, null]
+//     Expected result : [15, -22, 47]
+
+
+
+// 8 ### longestCommon
+// Write a JavaScript function to find the longest common starting substring in a set of strings.
+
+//     Sample array : console.log(longest_common_starting_substring(['go', 'google']));
+//     Expected result : "go"
+
+
+
+// 7 ### fillArray
+// Write a JavaScript function to fill an array with values (numeric, string with one character) on supplied bounds.
+
+// Test Data : 
+// console.log(num_string_range('a', "z", 2));
+// ["a", "c", "e", "g", "i", "k", "m", "o", "q", "s", "u", "w", "y"]
+
+
+
+// 6 ### removeElement
+// Write a JavaScript function to remove a specific element from an array.
+
+//     Test data :
+//     console.log(remove_array_element([2, 5, 9, 6], 5));
+//     [2, 9, 6]
+
+
+
+// 5 ### findArrayWithElement
+// Write a JavaScript function to find an array contains a specific element.
+
+//     Test data :
+//     console.log(remove_array_element([2, 5, 9, 6], 5));
+//     [2, 9, 6]
+
+
+
+// 4 ### emptyArray
+// Write a JavaScript script to empty an array keeping the original size.
+
+
+
+// 3 ### getNthLargest
+// Write a JavaScript function to get nth largest element from an unsorted array.
+
+//     Test Data :
+//     console.log(nthlargest([ 43, 56, 23, 89, 88, 90, 99, 652], 4));
+//     89
+
+
+
+// 2 ### array_filled (numeric)
+// Write a JavaScript function to create a specified number of elements with pre-filled numeric value array.
+
+//     Test Data :
+//     console.log(array_filled(6, 0)); 
+//     [0, 0, 0, 0, 0, 0]
+//     console.log(array_filled(4, 11));
+//     [11, 11, 11, 11]
+
+
+
+// 1 ### filterFalsy
+// Write a JavaScript function to filter false, null, 0 and blank values from an array.
+
+//     Test Data :
+//     console.log(filter_array_values([58, '', 'abcd', true, null, false, 0]));
+//     [58, "abcd", true]
+
+
+
+// HANGMAN
+
+// Definir un array con 20 palabras aleatorias (pueden estar relacionadas con una categoría como por ejemplo "Verduras", "Equipos de fútbol)
+// Seleccionar una palabra y definir el número de vidas del usuario
+// Utilizando el window.prompt debéis ir preguntando letras al usuario
+// Si la letra es correcta, debéis informarlo e irla escribiendo
+// Si la letras no es correcta debéis informarlo y descontar el número de vidas del usuario
+// Si se agotan las vidas del usuario, mostrar un mensaje "GAME OVER"
+// Si se adivina la palabra, mostrar un mensaje "YOU WIN"
+
+var greekGods = ['Achelous','Aeolus','Aether','Alastor','APOLLO','ARES','Aristaeus','Asclepius','ATLAS','Attis','Boreas','Caerus','Castor','Cerus','CHAOS','Charon','Cronos','Crios','CRONUS','Dinlas','DIONYSUS'];
```

Comment:
@cristiam86 Con hangman me he quedado bloqueada, cada funcion va por separado, pero al juntarlas en hangman() dice que wordEncripted is not defined y no veo por qué...